pages/blog/_posts/2020-10-01-what-is-short-messages.md

```diff
@@ -94,48 +102,108 @@ phonecalls. It was decided at a meeting in Oslo to be released to the
-messages. In the end (_hehe_) the `End` message is transmitted as a
-response and the transaction is finished.
-
-When one of these chunk (or the full message if lesser than 115 bytes)
-is delivered to the recipient, a response is sent back with the
-`MT-Forward-SM` containing a delivery status.
-
-OR, in the case that you are sending an *SMS* to a recipient in another
-network and the recipient is not available, it might contain the time
-the receiving *SMSC* will hold the message and retry to send it to the
-recipient.
+The *SMSC* then sends another packet, this time a `MT-Forward-SM`,
+towards the *MSC* in the recipients network. In this case *MT* stands
+for Mobile Terminated, meaning it goes towards the recipients phone.
+
+_Dia have amazing icons:_
+<div>
+    <img src="/img/blog/sms/forward-sm.svg" alt="You calling your mom" />
+</div>
+
+The similarities in *MO* and *MT* requests are that they both contain
+a origin and destination address as well as the user data (your actual
+text message), and a possibly a correlation id which is basically a
+mapping between your SIM-card id and a temporary id and was originally
+used for making sure that the sending network payed for *SMS*s towards
+the receiving network.
+
+For *MO* the origin address is your *MSISDN* (read telephone number),
+and the destination is the *GT* address (Global title; a way to route
+stuff) of the *SMSC*. For *MT* messages the origin address is the *GT*
+of the *SMSC* and the destination address is either the recipients
+*IMSI* (read SIM-card) or the recipients correlation id. It could also
+be a *LMSI* which is a 4-byte network location identifier if the
+recipient is also within the same network as the sender.
+
+Ever wondered why there is a limit to the size of the text message you
+are sending?
+
+<div class="left-right-row">
+    <div class="text">Two characters left on a GSM7 encoded SMS.</div>
+    <img class="image" src="/img/blog/sms/160_chars.png" alt="Characters left: 2/160" />
+</div>
+
+If you (god forbid) you would break the protocol and send a text
+message greater than 140 bytes, which translates to 160, 140, or 70
+characters depending on locale [1], then your phone would break up the
+message into multiple text messages. This arbitrary size of 140 bytes
+is not really arbitrary at all. It was chosen because it would
+precisely fit into a single *MTP3* *SIF* (Signalling Information
+Field) when routing label, *SCCP*, *TCAP* and *MAP* layers were taken
+into account.
+
+[1] There is something called *GSM7* bit-packing. Instead of using 1
+byte (8 bits) per character, *GSM7* uses 7 bits. This means that
+instead of 140 characters, you can get up to 160 characters per
+*SMS*. The drawback is that you will have a smaller subset of
+characters to use, only the most common is supported. If you include
+any non-*GSM7* characters in your *SMS* then the *SMS* will
+automatically be converted to use *USC-2* instead. *USC-2* uses
```

Comment:
Or on "newer" phones (standardized in 2008), on some locales, will detect and use a different charset with GSM7 shift tables. Covers Portuguese, Spanish, Turkish and 10 Indian languages, this where the 152 chars come in.